### ISSUE TITLE
LCFS - BUG - Other diesel

### ISSUE THREAD

**Issue by airinggov** [bcgov/lcfs#1553]
**Describe the Bug**  
The label 'Other diesel' should be updated to 'Other diesel fuel'. Additionally, it is currently resolving to 'Prescribed carbon intensity' when it should provide the select options for Default carbon intensity and Fuel codes. The volume is being incorrectly added to Line 1 of the summary instead of Line 2, as 'Other diesel fuel' is a renewable fuel.

**Expected Behaviour**  
- The label should display as 'Other diesel fuel'.  
- 'Determining carbon intensity' should provide the select options for Default carbon intensity and Fuel codes
- ~~Volumes should be added to Line 2 of the summary under Diesel.~~ **(may already be resolved other card)**

**Actual Behaviour**  
- The label shows as 'Other diesel'.  
- 'Determining carbon intensity' incorrectly resolves to 'Prescribed carbon intensity'.  
- ~~Volumes are incorrectly added to Line 1 of the summary.~~

**Implications**  
- Incorrect labeling and resolution lead to data misrepresentation.  
- Renewable fuel volumes are inaccurately categorized, potentially affecting reporting and compliance calculations.  

**Steps To Reproduce**  
User/Role: BCeID user.  
1. Navigate to the relevant data entry, Fuel supply, Export fuel, Other use, Allocation agreements 
2. Input Fuel type 'Other diesel'.  
3. Observe that:  
   - The label reads 'Other diesel'.  
   - It resolves to 'Prescribed carbon intensity'.  
   - Volumes are added to Line 1 of the summary instead of Line 2.

**Comment by airinggov** [comment]
Hi Arturo, this critical 2 point card has been in progress for 2 weeks, is it near completion?

**Comment by areyeslo** [comment]
sorry Al in the last 2 weeks I worked just 3 days, and compleated the other ticket in WIP 1540 and 1509. This ticket is the next. I will let you know when it is ready for your review.

**Comment by airinggov** [comment]
@areyeslo ok thanks, I guess we should make a point that if taking a critical card it can be completed before vacation or else assigned to another dev

**Comment by areyeslo** [comment]
sorry it is my bad. I preferred to finish the other implementations because they were more complex. I lost the priority of the tickets, last time happening this, I will keep on mind.

**Comment by areyeslo** [comment]
We have the following scenario:
fuel export &  supply of fuel are using `get_fuel_supply_options` function which includes the following logic:
For non-fossil derived fuels includes ALL provisions where provision of the act is not  `Prescribed carbon intensity - section 19 (a)` for other diesel option. Then, we are showing provision_of_the_act_id = 2 (Fuel code) and provision_of_the_act_id=3 (Default carbon intensity)

allocation agreement is using ``get_formatted_fuel_types` which is just querying by fuel type = `other diesel` and getting the provision_of_the_act_id value in `fuel_type` table which in dev is provision_of_the_act = 1. 

To make allocation_agreement the same we should remove the last logic and replace for the first logic.

**Comment by areyeslo** [comment]
Created PR for approval:
https://github.com/bcgov/lcfs/pull/1604

Note: January 6th I am off, I will be back on January 7th.

**Comment by airinggov** [comment]
The two select fields aren't being autoset or giving options to select

![Image](https://github.com/user-attachments/assets/83bc5d00-b480-43a9-9a37-88d7706e22e3)

**Comment by areyeslo** [comment]
@airinggov going to take a look.  the implementation for allocation agreement is using a different function which is isolated from the other schedules. Going to take a look at this.

**Comment by areyeslo** [comment]
On Dev, I am not getting the behaviour mentioned above (selecting fuel type - Other diesel fuel, auto sets Fuel Category and End use as well as CI of fuel).

Fuel Supply schedule - autoselects fuel category, fuel use and CI of fuel:
![Image](https://github.com/user-attachments/assets/968d19b2-5d89-4d19-a0b4-3d0cd000309c)

![Image](https://github.com/user-attachments/assets/abeb61c2-4739-45ff-a5fb-1320cd046efe)


Allocation Agreement -  autoselects fuel category, fuel use and CI of fuel:
![Image](https://github.com/user-attachments/assets/7d49b8a4-ce57-4ba3-928b-a22bbec9df11)


Fuel Export is NOT selecting automatically End use:
![Image](https://github.com/user-attachments/assets/d9c49446-e4b5-4d98-8d4f-581e666d4c65)

**Comment by areyeslo** [comment]
These are the PR related in the comments, but there is no affectation between them:
https://github.com/bcgov/lcfs/pull/1587/files
https://github.com/bcgov/lcfs/pull/1604/files

**Comment by airinggov** [comment]
@areyeslo Yes there was a conversation in Teams that I think addressed these issues through a data reset in dev

**Comment by airinggov** [comment]
@areyeslo looks great now
